dev-docs/RFCs/v8.0/luma-refactor.md

```diff
@@ -0,0 +1,83 @@
+- 14 separate modules with no clear sense of how they're meant to interoperate
+- 201 exported classes and functions, again without a clear sense of their relationships
+- A large code base. ~733kB of source files - compare to e.g. a full 3D engine like three.js at ~1.2MB (1.6x), a low-level library like PicoGL at ~183kB (0.25x)
+- Several unfinished or unused APIs that make it hard to tell what's reliable
+- Documentation is sprawling and not focused on the API. Examples demonstrate random scenes rather than API features
+
+Clearly defining luma.gl's goals as a framework, and structuring the codebase and documentation around those goals, would benefit it in the following ways:
+- Reduce its complexity and associated maintenance costs
+- Reduce the weight it imposes on applications that use it
+- Make its value proposition clear to potential users
+- Simplify adoption
+
+## Overview
+
+The first step towards simplifying luma.gl is to be explicit about what its goals are. I propose that changes be made around the following key principles:
+- Targeting intermediate to advanced WebGL users
+- All systems implemented should solve explicitly identified problems. Experiments are encouraged, but those that fail to find a use case should be removed
+- Structured around levels of abstraction, primarily helper tools for low-level WebGL work, WebGL wrapper classes, and higher-level engine features
+- Increased focus on architectural fundamentals, simplicity, building systems that are easy to reason about. Less focus on using the latest APIs or frameworks
+
+
+The actual refactor of luma.gl would occur in three phases:
+
+### Clean Out Unused Code
+
+This will warrant some discussion. The following are systems I believe should be considered for removal or consolidation:
+- [ModelNode](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/scenegraph/nodes/model-node.js), [BaseModel](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/base-model.js), [Model](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/model.js) - functionality could be supported with just the `Model` and `Group` classes
+- [CameraNode](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/scenegraph/nodes/camera-node.js) - Appears unused
+- [Uniform animations](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/base-model.js#L280-L333) - Complex and unused
+- [ShaderCache](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lib/shader-cache.js) - Superseded by `ProgramManager`
+- [core/lighting](https://github.com/uber/luma.gl/blob/7.2-release/modules/core/src/lighting/light-source.js), [core/materials](https://github.com/uber/luma.gl/tree/7.2-release/modules/core/src/materials) - currently empty or simple data classes
+- [core/multipass](https://github.com/uber/luma.gl/tree/7.2-release/modules/core/src/multipass) - Appears unused (has it been superseded by deck effects?)
```

Comment:
I always intended for deck.gl to use a render pass system from luma.gl (initially I expected that to be an evolution of this system) and I still think it should (perhaps an evolution of your abstract draw engine?)

I am not entirely against cutting the renderpass system, that said it is used in the `glfx` demo application, which is fairly impressive. 

Also without keeping at least some of the renderpass classes, the big library of postprocessing effects can not be used directly with luma.

Perhaps we could prune the system down to some minimal class needed to use the effects.